src/Microsoft.AspNet.Mvc.Xml/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -142,7 +150,32 @@ protected virtual Type GetSerializableType([NotNull] Type declaredType)
         /// <returns>The <see cref="DataContractSerializer"/> used during deserialization.</returns>
         protected virtual DataContractSerializer CreateSerializer([NotNull] Type type)
         {
-            return new DataContractSerializer(type, _serializerSettings);
+            try
+            {
+                // If the serializer does not support this type it will throw an exception.
+                return new DataContractSerializer(type, _serializerSettings);
+            }
+            catch (Exception)
+            {
+                // We do not surface the caught exception because if CanRead returns
+                // false, then this Formatter is not picked up at all.
+                return null;
+            }
+        }
+
+        private DataContractSerializer GetCachedSerializer(Type type)
```

Comment:
A type's serializer could be cached 2 times (input and output formatters), but I prefer this as there is clean separation between input and output serializers instead of having a common cache...ex: probably there can be a scenario where a type's serializer has different settings for input than output...